lib/src/facts/aix/operating_system_resolver.cc

```diff
@@ -0,0 +1,68 @@
+#include <internal/facts/aix/operating_system_resolver.hpp>
+#include <internal/util/regex.hpp>
+#include <facter/facts/os.hpp>
+#include <facter/execution/execution.hpp>
+#include <facter/util/file.hpp>
+
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+using namespace facter::util;
+namespace execution = facter::execution;
+
+string getattr(string object, string field)
+{
+    string result;
+
+    // To get architecture, can/should we use getconf KERNEL_BITMODE, getconf HARDWARE_BITMODE?
+    execution::each_line(
+        "/usr/sbin/lsattr", {"-El", object, "-a", field},
+        [&](string& line) {
+            if (!line.empty()) {
+                vector<string> tokens;
+                istringstream iss(line);
+                // istream_iterator is a great way to tokenize your life
+                copy(istream_iterator<string>(iss),
```

Comment:
Is this something we should use more generally, and put in a utility? I think other places we just use something like `boost::split(tokens, line, boost::is_space(), boost::token_compress_on)`.